client/src/ui/modules/military/battle-view/utils.tsx

```diff
@@ -0,0 +1,102 @@
+import { ClientComponents } from "@/dojo/createClientComponents";
+import { ArmyInfo } from "@/hooks/helpers/useArmies";
+import { Battle, EternumGlobalConfig, Health, Troops as SdkTroops, TroopConfig } from "@bibliothecadao/eternum";
+import { ComponentValue } from "@dojoengine/recs";
+import { getTotalTroops } from "./BattleHistory";
+
+export const getChancesOfSuccess = (
+  attackerArmy: ArmyInfo | undefined,
+  defenderArmy: ArmyInfo | undefined,
+  troopConfig: ComponentValue<ClientComponents["TroopConfig"]["schema"]>,
+) => {
+  if (!attackerArmy || !defenderArmy) return 0;
+
+  const attackingArmyStrength = Number(
+    fullStrength(attackerArmy.troops, troopConfig) * percentageLeft(attackerArmy.health),
+  );
+  const defendingArmyStrength = Number(
+    fullStrength(defenderArmy.troops, troopConfig) * percentageLeft(defenderArmy.health),
+  );
+
+  const attackSuccessProbability =
+    attackingArmyStrength / Math.max(Number(attackingArmyStrength + defendingArmyStrength), 1);
+  return attackSuccessProbability;
+};
+
+function fullStrength(
+  troops: SdkTroops,
+  troopConfig: ComponentValue<ClientComponents["TroopConfig"]["schema"]>,
+): bigint {
+  const knightCount = BigInt(troops.knight_count);
+  const paladinCount = BigInt(troops.paladin_count);
+  const crossbowmanCount = BigInt(troops.crossbowman_count);
+  const totalKnightStrength = BigInt(troopConfig.knight_strength) * knightCount;
+  const totalPaladinStrength = BigInt(troopConfig.paladin_strength) * paladinCount;
+  const totalCrossbowmanStrength = BigInt(troopConfig.crossbowman_strength) * crossbowmanCount;
+
+  return totalKnightStrength + totalPaladinStrength + totalCrossbowmanStrength;
+}
+
+function percentageLeft(health: ComponentValue<ClientComponents["Health"]["schema"]>): bigint {
+  if (health.lifetime === 0n) {
+    return 0n;
+  }
+  return (health.current * BigInt(10_000)) / health.lifetime;
```

Comment:
is this precision?

Reply:
My bad, this was a percentage value. Changing right now